sonar-csharp-plugin/src/main/resources/documentation.md

```diff
@@ -16,12 +16,26 @@ Discover and update the C#-specific [properties](/analysis/analysis-parameters/)
 
 To analyze tool-generated code (e.g. WCF code generated by `SvcUtil.exe`, protobuf code generated by `protoc`, Swagger client code generated by `NSwag`) for a specific C# project, enable the "Analyze generated code" setting inside **Project > Administration > General Settings > C#**. By default, tool-generated code files are skipped from analysis.
 
+Generated code detection is based on file name, special comments and attributes. Currently recognized values are in [GeneratedCodeRecognizer.cs](https://github.com/SonarSource/sonar-dotnet/blob/master/analyzers/src/SonarAnalyzer.Common/Helpers/GeneratedCodeRecognizer.cs).
+
 ## Scanner compatibility
 
 To analyze C# code, you need to use the [SonarScanner for .NET](/analysis/scan/sonarscanner-for-msbuild/) version 4.x or newer.
 
+## Exclusions
+
+Files excluded from analysis should be set in a project configuration. This exclusion still performs the analysis on the file and filters final results.
+
+As an alternative, `.editorconfig` file can be used to disable analysis for a specific rule on a file or directory. This can solve performance problems on large files.
+
+```
+[Path/File.cs]
+dotnet_diagnostic.Sxxx.severity = none
+```
+
```

Comment:
```suggestion

You can read more in the [official documentation](https://docs.microsoft.com/en-us/visualstudio/code-quality/use-roslyn-analyzers?view=vs-2019#manually-configure-rule-severity-in-an-editorconfig-file).
```

Reply:
I intentionally avoided that, as it's VS/dotnet stuff that's not specific to our analyzer. And docs show a very easy way how to disable rules contrary your QP/QG settings.